db-client-java/src/main/java/com/eventstore/dbclient/ListPersistentSubscriptions.java

```diff
@@ -0,0 +1,89 @@
+                result.completeExceptionally(new UnsupportedFeature());
+                return result;
+            }
+
+            if (args.supportFeature(FeatureFlags.PERSISTENT_SUBSCRIPTION_MANAGEMENT)) {
+                Persistent.ListReq.Options.Builder optionsBuilder = Persistent.ListReq.Options.newBuilder();
+
+                if (stream.equals("")) {
+                    optionsBuilder.setListAllSubscriptions(Shared.Empty.getDefaultInstance());
+                } else if (stream.equals("$all")) {
+                    optionsBuilder.setListForStream(Persistent.ListReq.StreamOption.newBuilder().setAll(Shared.Empty.getDefaultInstance()));
+                } else {
+                    optionsBuilder.setListForStream(Persistent.ListReq.StreamOption.newBuilder().setStream(Shared.StreamIdentifier.newBuilder().setStreamName(ByteString.copyFromUtf8(stream))));
+                }
+
+                Persistent.ListReq req = Persistent.ListReq.newBuilder()
+                        .setOptions(optionsBuilder)
+                        .build();
+
+                PersistentSubscriptionsGrpc.PersistentSubscriptionsStub stub = MetadataUtils
+                        .attachHeaders(PersistentSubscriptionsGrpc.newStub(args.getChannel()), options.getMetadata());
+
+                stub.list(req, GrpcUtils.convertSingleResponse(result, resp -> {
+                    List<PersistentSubscriptionInfo> infos = new ArrayList<>();
+
+                    for (Persistent.SubscriptionInfo wire : resp.getSubscriptionsList()) {
+                        infos.add(parseInfoFromWire(wire));
+                    }
+
+                    return infos;
+                }));
+            } else {
+                String suffix = "";
+
+                if (!stream.equals("")) {
+                    suffix = String.format("/%s", urlEncode(stream));
+                }
+
+                HttpURLConnection http = args.getHttpConnection(options, client.settings, String.format("/subscriptions%s", suffix));
+                try {
+                    http.setRequestMethod("GET");
+
+                    Exception error = checkForError(http.getResponseCode());
+                    if (error != null) {
+                        result.completeExceptionally(error);
+                    } else {
+                        String content = readContent(http);
+                        List<PersistentSubscriptionInfo> ps = new ArrayList<>();
+
+                        for (JsonNode node : getObjectMapper().readTree(content)) {
+                            ps.add(parseSubscriptionInfo(node));
+                        }
+                        result.complete(ps);
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                } finally {
+                    http.disconnect();
+                }
+            }
+            ;
```

Comment:
**NITPICK:** this could go to the line above 😉

Reply:
Done